component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/dao/JWTStorageManager.java

```diff
@@ -28,154 +28,194 @@
+                    long exp = rs.getTime(EXP_TIME,
+                            Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC))).getTime();
+                    long created = rs.getTime(TIME_CREATED,
+                            Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC))).getTime();
+                    JWTEntries.add(new JWTEntry(exp, created, tenantID));
+                }
+            } else {
+                prepStmt = dbConnection.prepareStatement(Util.getDBQuery(GET_JWT));
+                prepStmt.setString(1, jti);
+                rs = prepStmt.executeQuery();
+                while (rs.next()) {
+                    long exp = rs.getTime(EXP_TIME,
+                            Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC))).getTime();
+                    long created = rs.getTime(TIME_CREATED,
+                            Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC))).getTime();
+                    JWTEntries.add(new JWTEntry(exp, created));
+                }
             }
+
         } catch (SQLException e) {
             if (log.isDebugEnabled()) {
-                log.debug("Error when retrieving the JWT ID: " + jti, e);
+                log.debug("Error when retrieving the JWT ID: " + jti + " tenant id: " + tenantId, e);
             }
             throw new OAuthClientAuthnException("Error occurred while validating the JTI: " + jti + " of the " +
-                                                "assertion.", OAuth2ErrorCodes.INVALID_REQUEST);
+                    "assertion.", OAuth2ErrorCodes.INVALID_REQUEST);
         } finally {
             IdentityDatabaseUtil.closeAllConnections(dbConnection, rs, prepStmt);
         }
-        return jwtEntry;
+        return JWTEntries;
     }
 
     /**
      * To persist unique id for jti in the table.
      *
-     * @param jti         jti a unique id
-     * @param expTime     expiration time
-     * @param timeCreated jti inserted time
+     * @param jti         JTI a unique id.
+     * @param tenantId    Tenant id.
+     * @param expTime     Expiration time.
+     * @param timeCreated JTI inserted time.
      * @throws IdentityOAuth2Exception
      */
-    public void persistJWTIdInDB(String jti, long expTime, long timeCreated) throws OAuthClientAuthnException {
+    public void persistJWTIdInDB(String jti, int tenantId, long expTime, long timeCreated) throws OAuthClientAuthnException {
 
         Connection connection = null;
         PreparedStatement preparedStatement = null;
         try {
             connection = IdentityDatabaseUtil.getDBConnection();
             if (JWTServiceDataHolder.getInstance().isPreventTokenReuse()){
-                preparedStatement = connection.prepareStatement(SQLQueries.INSERT_JWD_ID);
+                preparedStatement = connection.prepareStatement(Util.getDBQuery(INSERT_JWD_ID));
+                if (Util.isIsTenantIdColumnIsAvailableInIdnOidcAuthTable()) {
```

Comment:
Can we take out the common set statements to the outside of if and else to avoid code duplicates? Need to adjust the query by changing the order.

Reply:
+1